docs/Displaying a User Interface/Main Screen Templates/index.md

```diff
@@ -1,76 +1,83 @@
 ```java
-SetDisplayLayout setDisplayLayoutRequest = new SetDisplayLayout()
-    .setDisplayLayout(PredefinedLayout.GRAPHIC_WITH_TEXT.toString());
-
-setDisplayLayoutRequest.setOnRPCResponseListener(new OnRPCResponseListener() {
+TemplateConfiguration templateConfiguration = new TemplateConfiguration().setTemplate(PredefinedLayout.GRAPHIC_WITH_TEXT.toString());
+sdlManager.getScreenManager().changeLayout(templateConfiguration, new CompletionListener() {
     @Override
-    public void onResponse(int correlationId, RPCResponse response) {
-        if (((SetDisplayLayoutResponse) response).getSuccess()) {
-            Log.i("SdlService", "Display layout set successfully.");
+    public void onComplete(boolean success) {
+        if (success) {
+            DebugTool.logInfo(TAG, "Layout set successfully");
         } else {
-            Log.i("SdlService", "Display layout request rejected.");
+            DebugTool.logInfo(TAG, "Layout not set successfully");
         }
     }
 });
-
-sdlManager.sendRPC(setDisplayLayoutRequest);
 ```
 !@
 
 @![javascript]
 ```js
-// sdl_javascript_suite v1.1+
-const setDisplayLayoutRequest = new SDL.rpc.messages.SetDisplayLayout();
-setDisplayLayoutRequest.setDisplayLayout(SDL.rpc.enums.PredefinedLayout.GRAPHIC_WITH_TEXT);
-const response = await sdlManager.sendRpcResolve(alert);
-if (response.getSuccess()) {
-    console.log('Display layout set successfully.');
-} else {
-    console.log('Display layout request rejected.');
-}
-// thrown exceptions should be caught by a parent function via .catch()
-
-// Pre sdl_javascript_suite v1.1
-const setDisplayLayoutRequest = new SDL.rpc.messages.SetDisplayLayout();
-setDisplayLayoutRequest.setDisplayLayout(SDL.rpc.enums.PredefinedLayout.GRAPHIC_WITH_TEXT);
-const response = await sdlManager.sendRpc(setDisplayLayoutRequest).catch(function (error) {
-    // Handle Error
+//TODO
+```
+!@
+
+Template changes can also be batched with text and graphics updates.
+
+@![iOS]
+##### Objective-C
+```objc
+//TODO
+```
+
+##### Swift
+```swift
+//TODO
+```
+!@
+
+@![android, javaSE, javaEE]
+```java
+sdlManager.getScreenManager().beginTransaction();
+sdlManager.getScreenManager().setTextField1("Line of Text");
+sdlManager.getScreenManager().changeLayout(templateConfiguration, new CompletionListener() {
+    @Override
+    public void onComplete(boolean success) {
+        // This listener will be ignored, and will use the CompletionListener sent in commit.
+    }
 });
-if (response.getSuccess()) {
-    console.log('Display layout set successfully.');
-} else {
-    console.log('Display layout request rejected.');
-}
+sdlManager.getScreenManager().setPrimaryGraphic(<#SDLArtwork#>);
+sdlManager.getScreenManager().commit(new CompletionListener() {
+    @Override
+    public void onComplete(boolean success) {
+        if (success) {
+            Log.i(TAG, "Text, Graphic and Template changed successful");
+        }
+    }
+});
+```
+!@
+
+@![javascript]
+```js
+//TODO
```

Comment:
```suggestion
sdlManager.getScreenManager().beginTransaction();
sdlManager.getScreenManager().setTextField1("Line of Text");
// The promise returned by changeLayout will not resolve because it is part of a batch update, and the await operator should be avoided as a result
sdlManager.getScreenManager().changeLayout(templateConfiguration);
sdlManager.getScreenManager().setPrimaryGraphic(<#SDLArtwork#>);
const success = await sdlManager.getScreenManager().commit();
if (success) {
    console.log('Text, Graphic, and Template changed successful");
}
```